3scale/porta
# Implement lazy components in login page

**What this PR does / why we need it**:

Some pages may contain a couple of components but not show all of them. For instance the login page has a different component for every sub-page: password change, request new password, sign in and sign up. Each of these components have their own pack file, plus the one for login.

Lazy components are declared in the code but not imported (loaded) until they're actually needed. Therefore, with this approach we can combine all 5 packs into one and the browser will only request the proper component according to the sub-page the user is in.

HOWEVER, this may not need so important performance-wise, since the component in used will always be rendered. Lazy components are the most useful when **it can't be ensured a component will be used or not**. In the login page this is known.

**Which issue(s) this PR fixes** 

Generic-performance-issue-N

**Verification steps** 

Check the following pages:
* Login
* Sign up (need invitation token)
* Request password
* Change password (need link after requesting new password)
